gammapy/astro/darkmatter/profiles.py

```diff
@@ -34,11 +34,15 @@ def scale_to_local_density(self):
         scale = (self.LOCAL_DENSITY / self(self.DISTANCE_GC)).to_value("")
         self.parameters["rho_s"].value *= scale
 
-    def _eval_squared(self, radius):
-        """Squared density at given radius."""
-        return self(radius) ** 2
+    def _eval_squared(self, radius, separation):
+        """Squared density at given radius together with the substitution part"""
+        return (
+            self(radius) ** 2
+            * radius
```

Comment:
Is there a factor 2 missing? As in 2* r * rho^2 / denominator

Reply:
Yes you are right, there is a factor 2 missing, thanks :)